libs/transloco/src/lib/transloco.signal.ts

```diff
@@ -0,0 +1,159 @@
+import {
+  computed,
+  inject,
+  Injector,
+  isSignal,
+  runInInjectionContext,
+  signal,
+  Signal,
+} from '@angular/core';
+import { toObservable, toSignal } from '@angular/core/rxjs-interop';
+import { switchMap } from 'rxjs';
+
+import { TRANSLOCO_SCOPE } from './transloco-scope';
+import { TranslocoService } from './transloco.service';
+import {
+  HashMap,
+  TranslateObjectParams,
+  Translation,
+  TranslocoScope,
+} from './types';
+
+type ScopeType = string | TranslocoScope | TranslocoScope[];
+type TranslateSignalKey = string | string[] | Signal<string> | Signal<string>[];
+type translateObjectSignal =
+  | TranslateObjectParams
+  | Signal<string>
+  | Signal<string>[];
+type TranslateSignalRef<T> = T extends unknown[]
+  ? Signal<string[]>
+  : Signal<string>;
+type TranslateObjectSignalRef<T> = T extends unknown[]
+  ? Signal<Translation[]>
+  : Signal<Translation>;
+
+/**
+ * Gets the translated value of a key as Signal
+ *
+ * @example
+ * text = translateSignal('hello');
+ * textList = translateSignal(['green', 'blue']);
+ * textVar = translateSignal('hello', { variable: 'world' });
+ * textSpanish = translateSignal('hello', { variable: 'world' }, 'es');
+ * textTodosScope = translateSignal('hello', { variable: 'world' }, { scope: 'todos' });
+ *
+ * @example
+ * dynamicKey = signal('hello');
+ * dynamicParam = signal({ variable: 'world' });
+ * object = translateSignal(this.dynamicKey, this.dynamicParam);
+ *
+ */
+export function translateSignal<T extends TranslateSignalKey>(
+  key: T,
+  params?: HashMap,
+  lang?: ScopeType,
+  injector?: Injector,
+): TranslateSignalRef<T> {
+  injector ??= inject(Injector);
```

Comment:
We want to check the injection context if no injector is provided
```suggestion
if (!injector) {
 assertInInjectionContext(translateSignal)
}
injector ??= inject(Injector);
```